mailstore-connector/src/com/axonivy/connector/mailstore/MailStoreService.java

```diff
@@ -677,16 +706,74 @@ private static Folder openFolder(Store store, String folderName, int mode) throw
 	private static String getVar(String store, String var) {
 		return Ivy.var().get(String.format("%s.%s.%s", MAIL_STORE_VAR, store, var));
 	}
+	
+	private static boolean isBasicAuth(String store) {
+		String auth = Ivy.var().get(String.format("%s.%s.%s", MAIL_STORE_VAR, store, AUTH));
+		return auth.equals("basic");
+	}
+	
+	private static Form buildForm(String store) {
+		Form form = new Form();
+		form.param("client_id", getVar(store, APP_ID));
+		form.param("client_secret", getVar(store, SECRET_KEY));
+		form.param("scope", getVar(store, SCOPE));
+		form.param("grant_type", getVar(store, GRANT_TYPE));
+		
+		return form;
+	}
+	
+	private static String getToken(String store) {
+		Form form = buildForm(store);
+
+		String tenantId = getVar(store, TENANT_ID);
+		String tokenUrlPrefix = getVar(store, "tokenUrl.tokenUrlPrefix");
+		String tokenUrlSuffix = String.format(getVar(store, "tokenUrl.tokenUrlSuffix"), tenantId);
+
+		if (StringUtils.isBlank(tokenUrlPrefix) || StringUtils.isBlank(tokenUrlSuffix)) {
+			LOG.error("url to get token cannot be null or empty");
+			throw buildError("getToken").withMessage("url to get token cannot be null or empty").build();
+		}
+		
+		TokenDTO result = null;
+		BpmError error = null;
+		SubProcessCallResult callResult = SubProcessCall.withPath("OAuth2Feature").withStartName("getToken")
+				.withParam("form", form)
+				.withParam("tokenUrlPrefix", tokenUrlPrefix)
+				.withParam("tokenUrlSuffix", tokenUrlSuffix)
+				.call();

```

Comment:
@nvvi I would refer REST request directly in java:
`
Ivy.rest().client("Retrieve Token").post(...)
`
It makes the code small: no need the SubProcess & Data class.
Refs: https://github.com/axonivy-market/demo-projects/blob/2ffca46ad9a3a6cc155b903c7881e3098d29e4e8/connectivity/connectivity-demos/src/com/axonivy/connectivity/rest/client/PersonClient.java#L53